### PR TITLE
Correct conceptual addresses/wallet mixup

### PR DESCRIPTION
**Pull request by anchpop**
Before, Wallet.t referred to a hash, while Address.t referred to a public key. This is a conceptual mixup - Wallet.t should really be a private key, and Address.t should really be a hash of a public key. This hopefully resolves that issue. 

Draft for now because I'm going to give it another look tomorrow to make sure I didn't miss anything/there's nothing in this PR that should be separated.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -10,6 +10,6 @@ Sometimes you will see it being used even for single cases `v |> f`, this is mor
 
 ### t
 
-A type named t means the main type of a module so, `Validators.t` can be read as `validators` and `Wallet.t` as `wallet`.
+A type named t means the main type of a module so, `Validators.t` can be read as `validators` and `Address.t` as `wallet`.
```
Comment (EduardoRFS) [inline HACKING.md:13]: hmm no?

File: protocol/wallet.rei [hunk 2]
```diff
@@ -1,14 +1,26 @@
 open Mirage_crypto_ec;
-open Helpers;
 
 [@deriving (ord, yojson)]
 type t;
+[@deriving (ord, yojson)]
+type pub_;
+
+let genesis_key: t;
+let genesis_pubkey: pub_;
```
Comment (EduardoRFS) [inline protocol/wallet.rei:9]: Let's go with `secret` for private keys and `key` to public keys. This is how it's used in michelson and it also makes clear, `secret` is a secret.

File: protocol/wallet.rei [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
+
+let wallet_of_privkey: Ed25519.priv => t;
+let wallet_to_privkey: t => Ed25519.priv;
+let pub_of_Ed25519pub: Ed25519.pub_ => pub_;
+let pub_to_Ed25519pub: pub_ => Ed25519.pub_;
```
Comment (EduardoRFS) [inline protocol/wallet.rei:14]: If they're alias and you will export the type like this, it's better to just export them with as `t = pri Ed25519.priv`

File: protocol/wallet.rei [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
+
+let pubkey_of_wallet: t => pub_;
+let make_pair: unit => (t, pub_);
+let make_pubkey: unit => pub_;
+
+let to_hex: string => string;
+let of_hex: string => string;
```
Comment (EduardoRFS) [inline protocol/wallet.rei:21]: Pretty sure of_hex cannot have this type and should not be exported here, currently we avoid exception as we don't have an effect system.

File: protocol/wallet.rei [hunk 5]
```diff
@@ -1,14 +1,26 @@
...
-let get_pub_key: Address.key => Address.t;
-let make_wallet: unit => (Ed25519.priv, t);
-let make_address: unit => t;
-let address_to_blake: t => BLAKE2B.t;
-let address_of_blake: BLAKE2B.t => t;
-let address_to_string: t => string;
+/*
+ let compare_pub: (pub_, pub_) => int;
+ let compare: (t, t) => int;
+ */
```
Comment (EduardoRFS) [inline protocol/wallet.rei:26]: If it's not used anymore remove it

File: protocol/wallet.re [hunk 6]
```diff
@@ -1,29 +1,66 @@
...
-let address_to_string = wallet =>
-  wallet |> address_to_blake |> BLAKE2B.to_string;
+let of_hex = hex => Hex.to_string(`Hex(hex));
+
+let to_yojson = t =>
+  `String(Ed25519.priv_to_cstruct(t) |> Cstruct.to_string |> to_hex);
+let of_yojson =
+  fun
+  | `String(key) =>
+    // TODO: this raises an exception
```
Comment (EduardoRFS) [inline protocol/wallet.re:28]: This is probably an outdated comment

File: protocol/wallet.rei [hunk 7]
```diff
@@ -1,14 +1,26 @@
...
+
+let genesis_key: t;
+let genesis_pubkey: pub_;
+
+let wallet_of_privkey: Ed25519.priv => t;
+let wallet_to_privkey: t => Ed25519.priv;
+let pub_of_Ed25519pub: Ed25519.pub_ => pub_;
+let pub_to_Ed25519pub: pub_ => Ed25519.pub_;
+
+let pubkey_of_wallet: t => pub_;
```
Comment (EduardoRFS) [inline protocol/wallet.rei:16]: `to_pubkey` or `pubkey`, you will always read it with the module as `Wallet.pubkey` or `Wallet.to_pubkey`

**Review by EduardoRFS — CHANGES_REQUESTED** [review]

File: protocol/wallet.re (hunk 6 above)
Comment (anchpop) [reply protocol/wallet.re:28]: Thought so too, just didn't remove it to keep this PR small https://github.com/marigold-dev/sidechain/blob/main/protocol/address.re#L16

File: protocol/wallet.rei (hunk 4 above)
Comment (anchpop) [reply protocol/wallet.rei:21]: What type does it have here? https://github.com/marigold-dev/sidechain/blob/main/protocol/address.re#L10 my editor says `string => string`. Probably right that it shouldn't be exported

File: protocol/wallet.rei (hunk 3 above)
Comment (anchpop) [reply protocol/wallet.rei:14]: Should I be using the `:>` syntax>

File: protocol/wallet.rei (hunk 2 above)
Comment (anchpop) [reply protocol/wallet.rei]: Does this mean we should use `Wallet.secret` instead of `Wallet.t`?

Comment (EduardoRFS) [reply protocol/wallet.rei]: A wallet is always a secret, saying `Wallet.secret` would be redundant.

File: protocol/wallet.rei (hunk 4 above)
Comment (EduardoRFS) [reply protocol/wallet.rei]: Yes, but we handle the exception internally

at > https://github.com/marigold-dev/sidechain/blob/main/protocol/address.re#L18

**Comment by d4hines** [comment]
Hey @anchpop, can you rebase or close this PR?

**Pull request closed without merging**